pkg/pulumiyaml/ast/expr.go

```diff
@@ -375,8 +375,12 @@ func tryParseAssetOrArchive(k, v Expr) (Expr, syntax.Diagnostics, bool) {
 	if fn, ok := assetOrArchiveFunctions[fnName.Value]; ok {
 		s, ok := v.(*StringExpr)
 		if !ok {
-			diags.Extend(syntax.NodeError(v.Syntax(), fmt.Sprintf("The argument to %s must be a string literal", fnName.Value), ""))
-			return nil, diags, true
+			i, ok := v.(*InterpolateExpr)
+			if !ok {
+				diags.Extend(syntax.NodeError(v.Syntax(), fmt.Sprintf("The argument to %s must be a string literal or interpolated expression", fnName.Value), ""))
+				return nil, diags, true
+			}
+			return fn(k.Syntax(), fnName, String(i.Syntax().String())), diags, true
```

Comment:
I think this will result in the string containing the (un-interpolated) `"${...}`. 

Can you add a unit test like the one here https://github.com/pulumi/pulumi-yaml/blob/983b9a7894b7c18ed7c2cc49665c1cd55e2f5a6f/pkg/pulumiyaml/run_test.go#L354